### PR TITLE
refactor(Core/DB): Deprecate MariaDB 10.4 and below

### PR DESCRIPTION
**Pull request by Kitzunu**
## Changes Proposed:
- Deprecate older versions of MariaDB as they cannot run queries that MySQL 5.7 and 8.0 can run.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/database/Database/DatabaseWorkerPool.cpp [hunk 1]
```diff
@@ -40,8 +40,8 @@
 #endif
 
 #if MARIADB_VERSION_ID >= 100600
-#define MIN_MYSQL_SERVER_VERSION 100200u
-#define MIN_MYSQL_CLIENT_VERSION 30203u
+#define MIN_MYSQL_SERVER_VERSION 100500u
+#define MIN_MYSQL_CLIENT_VERSION 30503u
```
Comment (Winfidonarleyan) [inline src/server/database/Database/DatabaseWorkerPool.cpp:44]: I'm not sure about this line.

**Review by Winfidonarleyan — APPROVED** [review]

File: src/server/database/Database/DatabaseWorkerPool.cpp (hunk 1 above)
Comment (Kitzunu) [reply src/server/database/Database/DatabaseWorkerPool.cpp:44]: me neither, but I could not find where it is located

Comment (Winfidonarleyan) [reply src/server/database/Database/DatabaseWorkerPool.cpp:44]: in `MariaDB 10.6` only

**Comment by Winfidonarleyan** [comment]
And need wiki changes

**Comment by Kitzunu** [comment]
> And need wiki changes

There is no info bout it on the wiki, but I will fix it

File: src/server/database/Database/DatabaseWorkerPool.cpp (hunk 1 above)
Comment (Kitzunu) [reply src/server/database/Database/DatabaseWorkerPool.cpp:44]: then I will just change back client thing? Or do you have time to test it fast with your setup?

I only run mysql

**Pull request merged**